transcom/mymove
# MAIN - B-21415 Prime Sim update for payment requests

## [B-21415](https://www13.v1host.com/USTRANSCOM38/assetdetail.v1?number=B-21415)

INT PR: https://github.com/transcom/mymove/pull/14321

## Summary

This allows the Prime to submit a payment request for international crate/uncrate. Also added more crating details to the UI in the Prime sim - payment request page.

- Added `ICRT` and `IUCRT` to allow payment requests
- Added `Market`, `Item size`, `Crate size`, `External crate` fields to the UI in payment request page.
- Now shows `Market` field under the international crate service item - Prime api - GET moveTaskOrder.

### How to test

1. In your `.envrc` file -> change to `export FEATURE_FLAG_ENABLE_ALASKA=true` to turn on Alaska FF locally.
2. Close your server -> run `direnv allow` and then restart your server.
3. Create a move and get the the step where the Prime can see the move and is able to create a service item (or use `HHGMoveInSITEndsTomorrow` testharness.
4. In Prime Sim -> find the move.
5. Click `Add Service Item` button for the shipment.
6. Select `International Crating` for "Service item type".
7. Select `International Crating` for "Service item code".
8. You should see `Standalone Crate` and `External Crate` options for "International crating" service item code and not "International Uncrating".
9. Test all the options to verify that you are able to create those service items.
10. Make sure item length/width/height is less than the Crate length/width/height.
11. Find the move as TOO and go to move task order tab -> scroll down to requested service items table and approve.
12. In prime sim -> find the move -> click "Create payment request" and scroll down and verify you can see the details(`Market, item size, crate size, and external crate`) for International Crating or International Uncrating.
13. Click the "Add to payment` checkbox and submit payment request. Verify you get that success message.

## Screenshots
<img width="851" alt="image" src="https://github.com/user-attachments/assets/e978f023-3bb7-4b81-a4ef-9d7018320824">
<img width="807" alt="image" src="https://github.com/user-attachments/assets/85d3b626-cfe3-4859-a335-91330051a94b">
